File: haskell/009-special_pythagorean_triplet.hs
-- Special Pythagorean triplet
--
-- Problem 9
-- A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
--
-- a^2 + b^2 = c^2
-- For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
--
-- There exists exactly one Pythagorean triplet for which a + b + c = 1000.
-- Find the product abc.


main = do
    print (truncate (pyth_triplet 1 1))

pyth_triplet :: Float -> Float -> Float
pyth_triplet a b
    | a + b + c == 1000 = a * b * c
    | b >= 1000 = pyth_triplet (a + 1) 0
    | otherwise = pyth_triplet a (b + 1)
    where c = sqrt (a ^ 2 + b ^ 2)

File: julia/009-special_pythagorean_triplet.jl
###
# Special Pythagorean triplet
# Problem 9
#
# A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
#  a2 + b2 = c2
# For example, 32 + 42 = 9 + 16 = 25 = 52.
# There exists exactly one Pythagorean triplet for which a + b + c = 1000.Find the product
# abc.
###


using Base.Iterators


for c in countfrom(1)
    for b in 1:(c - 1)
        for a in 1:(b - 1)
            if  a ^ 2 + b ^ 2 == c ^ 2 && a + b + c == 1000
                println(a * b * c)
                exit(0)
            end
        end
    end
end


